ME2Randomizer/Classes/Randomizers/ME2/Misc/RTexts.cs

```diff
@@ -167,11 +172,324 @@ private static void UwuifyTalkFile(TalkFile tf, bool keepCasing, List<int> skipI
                 }
 
                 var str = sb.ToString();
-                str = str.Replace("fuck", keepCasing ? "UwU" : "uwu", StringComparison.InvariantCultureIgnoreCase);
+
+                if (addReactions)
+                {
+                    str = AddReactionToLine(strData, str, keepCasing);
+                }
+                else //reactions not enabled, use original uwuify UwU insertion
+                { 
+                    str = str.Replace("fuck", keepCasing ? "UwU" : "uwu", StringComparison.InvariantCultureIgnoreCase);
+                }
+
                 TLKHandler.ReplaceString(sref.StringID, str, langCode);
             }
         }
 
+        private static string AddReactionToLine(string vanillaLine, string modifiedLine, bool keepCasing)
+        {
+            string finalString = "";
+            bool dangerousLine = false;
+
+            if (modifiedLine.Length < 2 || Regex.Matches(modifiedLine, "[a-zA-Z]").Count == 0 || vanillaLine.Contains('{'))
+            {
+                //I should go.
+                return modifiedLine;
+            }
+
+            string dangerousCharacters = "<\n";
+            if (modifiedLine.IndexOfAny(dangerousCharacters.ToCharArray()) >= 0 || vanillaLine.Length > 200)
```

Comment:
this will be faster if the string is actually declared static, outside of the function, as char array. `private static char[] dangerousCharacters = new char[] { '<', '\\', \n' }`.

This prevents allocations, and also doesn't require a call to .ToCharArray().

Reply:
I messed with moving dangerousCharacters outside the function as a static and it actually slowed execution down in a tiny bit (1/5 of a second). Best guess is compiler optimization throwing me a bone. Or some weird behavior with a debug build. I just left it where it was. 

Did change it to a character array, though. Using a string that way was silly.